src/topp/DatabaseSuitability.cpp

```diff
@@ -209,35 +214,64 @@ class DatabaseSuitability :
     Size count_re_ranked = 0;
     Size count_interest = 0;
 
-    for (const auto& pep_id : pep_ids)
+    for (auto& pep_id : pep_ids)
     {
-      const vector<PeptideHit>& hits = pep_id.getHits();
+      vector<PeptideHit>& hits = pep_id.getHits();
+      bool q_value_score = (pep_id.getScoreType() == "q-value");
 
       if (hits.empty()) continue;
 
+      // sort hits by q-value
+      if (q_value_score)
+      {
+        sort(hits.begin(), hits.end(),
+          [](const PeptideHit& a, const PeptideHit& b)
+          {
+            return a.getScore() < b.getScore();
+          });
+      }
+      else
+      {
+        if (!hits[0].metaValueExists("q-value"))
+        {
+          throw(Exception::Precondition(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No q-value found at peptide identification nor at peptide hits. Make sure 'False Discovery Rate' is run beforehand."));
+        }
+
+        sort(hits.begin(), hits.end(),
```

Comment:
Ah yes you want to sort by q-value.. I think the ranking should be the same with the normal score. Then you can use pep_id.sort(). Which is more concise and a bit faster (since it does not access meta values). But yes, maybe this is a bit safer. Up to you since it is probably not a bottleneck.

Reply:
While the ranking should be the same for the score from which the q-value was calculated, I'm not sure that this is necessarily the case for any other score. That's why I decided to make sure the sorting is done by q-value.